### PR TITLE
Fix - Improve dependency management

### LINKED ISSUE #59: Mixed dependencies
What I see: Some dependencies are included through bower others without any dependency management (aka downloaded and included under /vendor/)
What I expect: The project to use the same pattern throughout, preferably using bower or another dependency manager for consistency. 

Possible Fix: Remove all JS depences and incle through bower, include bourbon and neat through bower, switch up gulpfile to match new pattern.

### PR DESCRIPTION
**Pull request by Sambillingham**
This branch works to bring our dependencies under control. closes #59 

Javascript and CSS dependencies are included through bower with the exception of  - 

Modernizer - The gulp Modernizr plugin is super alpha generates us a 50kb file when we are only using the media query check, this is 5kb download when building online with the custom build tool. (https://modernizr.com/download?mediaqueries-dontmin-mq-shiv&q=js%20medi) 
 
Selectivizr-min.js - included conditionally in the head for ie8 support  (no reason why this cant also be included through bower, also not 100% sure if jQuery needs to be defined before selectivizr)
- [x] Look into this before merging 

Should probably write up some more detailed info in the readme file before merging this, seperate issue for that at #42 

- [x] Bower info in readme
- [x] Info about why we need the Shim for non commonJs modules and how all that stuff works

### CODE REVIEW

**Comment by Sambillingham** [comment]
Normalize can and should also be added as a dependency through bower
- [ ] add normalize through bower

File: _layouts/default.html [hunk 1]
```diff
@@ -21,6 +21,11 @@
     {% else %}
     <link rel="stylesheet" href="/css/main.css">
     {% endif %}
+
+    <!--[if (gte IE 6)&(lte IE 8)]>
+      <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+      <script type="text/javascript" src="/js/vendor/selectivizr.js"></script>
```
Comment (Sambillingham) [inline _layouts/default.html]: Selectivizr is included by bower to allow Bourbon generated responsive CSS to work in ie8 - Selectivizr  depends on jQuery. Included in the head of each page within an IE conditional so it does not need to be bundled along with the other JS and served to everybody

**Comment by Sambillingham** [comment]
I've attempted to improve the read.me along with the changes that this branch brings working towards #42 could be clafied further.

**Comment by Sambillingham** [comment]
I'm going to merge this in now as our other branches are relying on this.

I am still not sold on the browserify/bower not working nicely together issues - [debowerify](https://github.com/eugeneware/debowerify) could solve some of these issues as would switching to system JS based bundler.
